pkg/utils/utils.go

```diff
@@ -334,3 +335,38 @@ func addResourceVersion(patchBytes []byte, resourceVersion string) ([]byte, erro
 	}
 	return versionBytes, nil
 }
+
+func GetMergedNamespacesWithLabelSelector(namespaceList []string, namespaceSelectors map[string]string) ([]string, error) {
+	if len(namespaceSelectors) == 0 {
+		return namespaceList, nil
+	}
+	uniqueNamespaces := make(map[string]bool)
+	for _, ns := range namespaceList {
+		uniqueNamespaces[ns] = true
+	}
+
+	for key, val := range namespaceSelectors {
+		namespaces, err := core.Instance().ListNamespaces(map[string]string{key: val})
+		if err != nil {
+			return nil, err
+		}
+		for _, namespace := range namespaces.Items {
+			uniqueNamespaces[namespace.GetName()] = true
+		}
+	}
+
+	migrationNamespaces := make([]string, 0, len(uniqueNamespaces))
+	for namespace := range uniqueNamespaces {
+		migrationNamespaces = append(migrationNamespaces, namespace)
+	}
+	return migrationNamespaces, nil
+}
+
+func IsSubList(list1 []string, list2 []string) bool {
```

Comment:
Although I think in our case it practically doesn't matter, the time complexity of this method is O(m*n), while we could instead use a map and do it in O(m+n) where m and n are lengths of list1 and list2.
I had written the below function as part of #1631 
```
// isSubset returns true if the first slice is subset of the second slice.
// If false it also returns the list of non subset strings.
func isSubset(listA []string, listB []string) (bool, []string) {
	nonSubsetStrings := make([]string, 0)
	superset := make(map[string]bool)
	for _, str := range listB {
		superset[str] = true
	}
	for _, str := range listA {
		if !superset[str] {
			nonSubsetStrings = append(nonSubsetStrings, str)
		}
	}
	return len(nonSubsetStrings) == 0, nonSubsetStrings
}
```